### PR TITLE
Caste Swap

### PR DESCRIPTION
**Pull request by Barnet2**
## About The Pull Request
Adds a new verb in the alien tab, Caste Swap
Every 15 minutes you can swap into another xeno inside of your tier, there is no timer when you join, only when you swap
Also makes Regress use evolution UI
## Why It's Good For The Game
No more having to evo and devo to get to another caste in your tier especially if slots are filled, plus more flexibility during sieges (probably the worst part of the game for Xenomorphs)
## Changelog
:cl:
balance: Caste Swap: Every 15 minutes a Xenomorph can change to another caste inside of their tier
code: Regress uses evo UI
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
this is very funny speedmergh

**Comment by aarrisu** [comment]
this is aweosme

**Comment by MiniMeatwad** [comment]
What.

**Comment by novaepee** [comment]
truly epic

**Comment by Zangeiti** [comment]
youve heard of shotgun juggling, now get ready for the caste wombo-combo juggle

**Comment by efzapa** [comment]
me omw to psychic crush, fling, then swap to rav and break all your bones in the span of 3 seconds

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 1]
```diff
@@ -10,58 +10,61 @@
...
-		else
-			CRASH("side_evolve() called without a valid tier")
-
-	var/list/castes_to_pick = list()
-	for(var/type in tiers_to_pick_from)
-		var/datum/xeno_caste/available_caste = GLOB.xeno_caste_datums[type][XENO_UPGRADE_BASETYPE]
-		castes_to_pick += available_caste.caste_name
-	var/castepick = tgui_input_list(src, "We are growing into a beautiful alien! It is time to choose a caste.", null, castes_to_pick)
-	if(!castepick) //Changed my mind
+	if(world.time - (GLOB.key_to_time_of_caste_swap[src.key] ? GLOB.key_to_time_of_caste_swap[src.key] : -INFINITY) < 9000) //casteswap timer, 15 minutes
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm:18]:
```suggestion
	if(world.time - (GLOB.key_to_time_of_caste_swap[key] ? GLOB.key_to_time_of_caste_swap[key] : -INFINITY) < 9000) //casteswap timer, 15 minutes
```

File: code/modules/mob/living/carbon/xenomorph/evolution.dm [hunk 2]
```diff
@@ -10,58 +10,61 @@
 
 	GLOB.evo_panel.ui_interact(src)
 
-/mob/living/carbon/xenomorph/verb/regress()
-	set name = "Regress"
-	set desc = "Regress into a lower form."
+/mob/living/carbon/xenomorph/verb/CasteSwap()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/evolution.dm:13]: No CamelCase, use snake_case

**Pull request merged**